Tests/SourceKitLSPTests/CallHierarchyTests.swift

```diff
@@ -447,4 +447,53 @@ final class CallHierarchyTests: XCTestCase {
       ]
     )
   }
+
+  func testIncomingCallHierarchyLooksThroughProtocols() async throws {
```

Comment:
I don't expect it to fail, but you could have a test for the class hierarchy equivalent as well. That could also have the case of `A -> B -> C` where `foo` is on each and callers of `B.foo()` shouldn't find `C.foo()`.

A test case for properties would be good too.

Reply:
Added a test case. Passed